turbinia/evidence.py

```diff
@@ -303,21 +313,24 @@ class GoogleCloudDiskRawEmbedded(GoogleCloudDisk):
   """
 
   def __init__(self, embedded_path=None, *args, **kwargs):
-    """Initialization for Google Cloud Disk."""
+    """Initialization for Google Cloud Disk containing a raw disk image."""
     self.embedded_path = embedded_path
     super(GoogleCloudDiskRawEmbedded, self).__init__(*args, **kwargs)
 
+    # This Evidence needs to have a GoogleCloudDisk as a parent
+    self.context_dependent = True
+
   def _preprocess(self):
-    self.local_path = google_cloud.PreprocessAttachDisk(self.disk_name)
-    self.loopdevice_path = mount_local.PreprocessLosetup(self.local_path)
+    # TODO, after we make all Disks Evidence try to mount their filesystem,
+    # the following mounting will be done in the parent evidence.
+    # and we'll set mount_path as the path to the mounted embedded disk image.
     self.mount_path = mount_local.PreprocessMountDisk(
-        self.loopdevice_path, self.mount_partition)
-    self.local_path = os.path.join(self.mount_path, self.embedded_path)
+        self.parent_evidence.device_path, 1)
```

Comment:
We never had an oracle for what partition is interesting.
Either tasks don't care because they use the device_path (ie: tasks using dfvfs internal magic) or we use the partition provided by the user on running turbiniactl
Though I don't remember why I removed self.mount_partition here, adding back

Reply:
Hmm I see the problem.

Unless we want to have an intermediary "SomethingPartitionEvidence" type (where device_path is guaranteed to point to a partition, and a Task will be used to generate these Partition type evidences from an input evidence), I would do something hacky like:
if the argument to PreprocessMountDisk() partition_number is None, assume the loopdevice_path (which might need to be renamed to blockdevice_path) is pointing to a partition, otherwise add p+partitio_number
WDYT?

Reply:
> Unless we want to have an intermediary "SomethingPartitionEvidence" type (where device_path is guaranteed to point to a partition, and a Task will be used to generate these Partition type evidences from an input evidence), 

Yeah, this is pretty much what I had planned for the DFVFS volume/partition enumeration, but I don't want to go down that route yet for this PR since it will change a lot of other things.

> I would do something hacky like:
> if the argument to PreprocessMountDisk() partition_number is None, assume the loopdevice_path (which might need to be renamed to blockdevice_path) is pointing to a partition, otherwise add p+partitio_number
> WDYT?

I don't think we should assume it points at a partition since it will rarely actually be doing that.  If we wanted to do something hacky I think we should either:
1) Try appending `1`, `p1` or `-part1` until we find a partition device.  
2) Use the losetup preprocessor so that we can be assured that we have the right partition device name.

#1 feels kind of hacky and might have other exceptions to the naming pattern that we'll need to keep updating, so maybe #2 will be better?

BTW, I think in general `loopdevice_path` should probably be just `device_path`.

Reply:
Okay for 2, the whole purpose of losetup is to create a path that follows '/dev/loop{some_number}p{partition_number_starting_from_one}'. Which is what the mounting preprocessor expects.

Reply:
> I could change the PreprocessMountDisk to take a list of paths to partition block devices, and use the provided partition number as an index in this list.
> This would mean changing PreprocessLosetup to return ['/dev/loop0p1', '/dev/loop0p2'] and google_cloud.PreprocessAttachDisk to return ['/dev/disk/by-id/google-persistent-disk0-part1', '/dev/disk/by-id/google-persistent-disk0-p2'. No 'guessing' is left to PreprocessMountDisk.

Yeah, this sounds good to me.  Thanks!